ReactNativeClient/lib/components/note-body-viewer.js

```diff
@@ -203,6 +210,33 @@ class NoteBodyViewer extends Component {
 		return this.forceUpdate_;
 	}
 
+	async onResourceLongPress(resourceId) {
+		const resource = await Resource.load(resourceId);
+		const name = resource.title ? resource.title : resource.file_name;
+
+		const action = await dialogs.pop(this, name, [
+			{ text: _('Share'), id: 'share' },
+		]);
+
+		if (action === 'share') {
+			const file_name = resource.file_name ?
+				`${resource.file_name}.${resource.file_extension}` :
+				resource.title;
+			const targetPath = `${Setting.value('resourceDir')}/${file_name}`;
```

Comment:
We probably shouldn't copy to the resource directory. I think there's a temp directory defined that you could use?

Reply:
I only see temp dir in FileApi: tempDirName. It seems to only be used for sync, not sure if using this is any better than resources dir.

Reply:
Also as I've just noticed WelcomeUtils aslo uses the resource dir for temp

Reply:
/storage/emulated/0/joplin/images
How about a directory like this?
There seems to be quite a number of applications to do this.

Reply:
Hmm, yes resourceDir is basically the document dir. I remember there was something a bit tricky about using a different dir for copying files which is why I've used it for welcome items too. So let's leave it in resourceDir for now then.